src/details/ArborX_DetailsDendrogram.hpp

```diff
@@ -89,33 +88,29 @@ template <typename ExecutionSpace, typename MemorySpace>
 void dendrogramUnionFind(ExecutionSpace const &exec_space,
                          Kokkos::View<WeightedEdge const *, MemorySpace> edges,
                          Kokkos::View<int *, MemorySpace> &parents,
-                         Kokkos::View<float *, MemorySpace> &parent_heights)
+                         Kokkos::View<float *, MemorySpace> &weights)
```

Comment:
I would prefer to leave `parent_heights` as the argument name, as it is unclear what the weights are from the function name. You can alias to `weights` inside, if you want.